functions.sh

```diff
@@ -260,3 +265,41 @@ remove_iscsi_target() {
         rm ${imgfile}
     fi
 }
+
+apply_updates_image() {
+    local image_url="${1}"
+    local updates_dir="${2}"
+
+    # Create the updates directory.
+    mkdir -p "${updates_dir}"
+
+    # Check the updates image.
+    if [[ -z "${image_url}" ]]; then
+        return
+    fi
+
+    # Download and extract the updates image.
+    ( cd "${updates_dir}" ; curl -f "${image_url}" | gzip -dc  | cpio -idu )
+}
+
+create_updates_image() {
+    local updates_dir="${1}"
+    local updates_img="${2}"
+
+    ( cd "${updates_dir}" ; find . | cpio -co | gzip -9 ) >"${updates_img}"
+}
+
+upload_updates_image() {
+    local tmp_dir="${1}"
+    local updates_img="${2}"
+
+    # Copy the updates image.
+    mkdir ${tmpdir}/http
+    cp "${updates_img}" ${tmpdir}/http/updates.img
+
+    # Start the http server.
+    start_httpd ${tmpdir}/http ${tmpdir}
```

Comment:
Maybe the test (perhaps via a function for updates image handling) should also stop the http server, as the tests starting it in .sh file do in cleanup() function.

Reply:
Fixed. However, why don't we do that by default?